### PR TITLE
Typecheck dev

### PR DESCRIPTION
**Pull request by imteazs**
Adding type checking to python code.

### CODE REVIEW

File: seqr/utils/logging_utils.py [hunk 1]
```diff
@@ -41,7 +42,7 @@ def formatMessage(self, record):
 
 class SeqrLogger(object):
 
-    def __init__(self, name=None):
+    def __init__(self, name: Optional[str]=None) -> None:
```
Comment (hanars) [inline seqr/utils/logging_utils.py:45]: According to PEP8, you need to add spaces around the equals signs now that you have added annotations: https://peps.python.org/pep-0008/#other-recommendations

> Don’t use spaces around the = sign when used to indicate a keyword argument, or when used to indicate a default value for an unannotated function parameter:
When combining an argument annotation with a default value, however, do use spaces around the = sign

**Review by hanars — CHANGES_REQUESTED** [review]

File: seqr/views/utils/permissions_utils.py [hunk 2]
```diff
@@ -1,58 +1,29 @@
...
-from settings import API_LOGIN_REQUIRED_URL, ANALYST_USER_GROUP, PM_USER_GROUP, INTERNAL_NAMESPACES, \
+    anvil_enabled, user_get_workspace_access_level, WRITER_ACCESS_LEVEL, OWNER_ACCESS_LEVEL,\
+    PROJECT_OWNER_ACCESS_LEVEL, CAN_SHARE_PERM
+from settings import API_LOGIN_REQUIRED_URL, ANALYST_USER_GROUP, PM_USER_GROUP, ANALYST_PROJECT_CATEGORY, \
     TERRA_WORKSPACE_CACHE_EXPIRE_SECONDS, SEQR_PRIVACY_VERSION, SEQR_TOS_VERSION, API_POLICY_REQUIRED_URL
+from typing import Callable, Optional
 
 logger = SeqrLogger(__name__)
 
-
```
Comment (hanars) [inline seqr/views/utils/permissions_utils.py:19]: I think this a merge error

File: seqr/views/utils/permissions_utils.py [hunk 3]
```diff
@@ -63,28 +34,28 @@ def _has_current_policies(user):
...
+def _require_permission(user_permission_test_func: Callable, error: str = 'User has insufficient permission') -> Callable:
     def test_user(user):
         if not user_permission_test_func(user):
             raise PermissionDenied(error)
         return True
     return test_user
 
 _active_required = user_passes_test(_require_permission(lambda user: user.is_active, error='User is no longer active'))
-def _current_policies_required(view_func, policy_url=API_POLICY_REQUIRED_URL):
+def _current_policies_required(view_func: Callable, policy_url: str=API_POLICY_REQUIRED_URL) -> Callable:
```
Comment (hanars) [inline seqr/views/utils/permissions_utils.py:45]: still missing spacing here, but otherwise looks good!

**Review by hanars — CHANGES_REQUESTED** [review]

**Comment by imteazs** [comment]
I might have accidentally overwrote my original work while trying to sync new changes, luckily I think I was able recreate them from monkeytype traces.

**Pull request merged**